exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -0,0 +1,1460 @@
+			ku.Name,
+			err)
+	}
+}
+
+// FetchTradablePairs returns a list of the exchanges tradable pairs
+func (ku *Kucoin) FetchTradablePairs(ctx context.Context, assetType asset.Item) (currency.Pairs, error) {
+	var cp currency.Pair
+	switch assetType {
+	case asset.Futures:
+		myPairs, err := ku.GetFuturesOpenContracts(ctx)
+		if err != nil {
+			return nil, err
+		}
+		pairs := make(currency.Pairs, 0, len(myPairs))
+		for x := range myPairs {
+			if strings.ToLower(myPairs[x].Status) != "open" { //nolint:gocritic // strings.ToLower is faster
+				continue
+			}
+			cp, err = currency.NewPairFromStrings(myPairs[x].BaseCurrency, myPairs[x].Symbol[len(myPairs[x].BaseCurrency):])
+			if err != nil {
+				return nil, err
+			}
+			pairs = pairs.Add(cp)
+		}
+		configFormat, err := ku.GetPairFormat(asset.Futures, false)
+		if err != nil {
+			return nil, err
+		}
+		return pairs.Format(configFormat), nil
+	case asset.Spot, asset.Margin:
+		myPairs, err := ku.GetSymbols(ctx, "")
+		if err != nil {
+			return nil, err
+		}
+		pairs := make(currency.Pairs, 0, len(myPairs))
+		for x := range myPairs {
+			if !myPairs[x].EnableTrading {
+				continue
+			}
+			cp, err = currency.NewPairFromString(strings.ToUpper(myPairs[x].Name))
+			if err != nil {
+				return nil, err
+			}
+			pairs = pairs.Add(cp)
+		}
+		return pairs, nil
+	default:
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, assetType)
+	}
+}
+
+// UpdateTradablePairs updates the exchanges available pairs and stores
+// them in the exchanges config
+func (ku *Kucoin) UpdateTradablePairs(ctx context.Context, forceUpdate bool) error {
+	assets := ku.GetAssetTypes(true)
+	for a := range assets {
+		pairs, err := ku.FetchTradablePairs(ctx, assets[a])
+		if err != nil {
+			return err
+		}
+		if len(pairs) == 0 {
+			return fmt.Errorf("%v; no tradable pairs", currency.ErrCurrencyPairsEmpty)
+		}
+		err = ku.UpdatePairs(pairs, assets[a], false, forceUpdate)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// UpdateTicker updates and returns the ticker for a currency pair
+func (ku *Kucoin) UpdateTicker(ctx context.Context, p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	p, err := ku.FormatExchangeCurrency(p, assetType)
+	if err != nil {
+		return nil, err
+	}
+	if err := ku.UpdateTickers(ctx, assetType); err != nil {
+		return nil, err
+	}
+	return ticker.GetTicker(ku.Name, p, assetType)
+}
+
+// UpdateTickers updates all currency pairs of a given asset type
+func (ku *Kucoin) UpdateTickers(ctx context.Context, assetType asset.Item) error {
+	switch assetType {
+	case asset.Futures:
+		ticks, err := ku.GetFuturesOpenContracts(ctx)
+		if err != nil {
+			return err
+		}
+		pairs, err := ku.GetEnabledPairs(asset.Futures)
+		if err != nil {
+			return err
+		}
+		for x := range ticks {
+			pair, err := currency.NewPairFromString(ticks[x].Symbol)
+			if err != nil {
+				return err
+			}
+			if !pairs.Contains(pair, true) {
+				continue
```

Comment:
Sorry sam, this impacts more than just the websocket. Please go over all Futures based wrapper and websocket work to ensure that currency pairs are being correctly sorted
<img width="1166" alt="image" src="https://github.com/thrasher-corp/gocryptotrader/assets/9261323/683eadb1-6851-4210-9178-f63fc9c3f066">

Reply:
In addition. With using DeriveFrom in a context like this, you'll need to do the following:
```go
			pair, err := pairs.DeriveFrom(ticks[x].Symbol)
			if err != nil {
				if errors.Is(err, currency.ErrPairNotFound) {
					continue
				}
				return err
			}
```
Otherwise the batch will fail

Reply:
I have changed all occurrence of NewPairFromString to DeriveFrom